### PR TITLE
feat(withdrawals): export tax evidence csv

### PR DESCRIPTION
**Pull request by adelphi-liong**
## Summary

- add an admin-only `GET /api/v1.0/Withdrawal/export` endpoint using the existing withdrawal filters without caller pagination
- keyset-page fully hydrated withdrawals in stable ascending `(CreatedAt, Id)` order, backed by a composite database index, and stream the complete result as UTF-8 RFC 4180 CSV
- export legacy completed withdrawals correctly as `fee = 0.00` and `net_amount = gross` when their pre-fee rows have no payout record; no-payment and unsettled statuses retain blank money cells
- keep the 7-day presigned `receipt_url` as a convenience while appending a durable `receipt_key` as column 23 for archived evidence
- emit stored payout fields and index-aligned card-refund evidence, with Asia/Singapore timestamps and spreadsheet-formula protection
- abort incomplete responses on repository or receipt-presigning failures, and expose `Content-Disposition` cross-origin so the frontend can use the server-selected filename

## Verification

- `dotnet test UnitTest/UnitTest.csproj --no-build --no-restore --filter 'FullyQualifiedName~Withdrawals'` — 206 passed
- `dotnet test UnitTest/UnitTest.csproj --no-build --no-restore` — 831 passed; 29 pre-existing timezone-data failures, all in `BookingServicePrioritizeTests`
- `dotnet build App/App.csproj --no-restore -v:minimal` — 0 errors
- `dotnet ef migrations has-pending-model-changes --project ./App --no-build` — no pending model changes
- `pre-commit run treefmt --all-files` — passed
- `git diff --check` — passed

## Export contract

```text
withdrawal_id,created_at,completed_at,status,method,user_id,user_email,gross_amount,fee,net_amount,paynow_number,airwallex_confirmation,payout_attempt,reconcile_attempts,completer_id,completion_note,receipt_url,refund_payment_intent_ids,refund_airwallex_ids,refund_amounts,refund_statuses,refund_settled_ats,receipt_key
```

## Notes

- `receipt_url` uses the MinIO/S3 7-day presign maximum; expiry inputs are bounded to the supported 1-second–7-day range, while `receipt_key` is the permanent object reference that can be re-signed later.
- Each export page and its refund evidence now load in one SQL statement; the `(CreatedAt, Id)` migration keeps repeated keyset pages index-served.
- Formula-like field values are prefixed with a literal apostrophe for Excel/Sheets safety.
- Timestamps are emitted as ISO 8601 Asia/Singapore values with an explicit `+08:00` offset, matching the `After`/`Before` calendar-day boundaries.
- The shared list/export validator accepts non-negative `Max`; invalid list statuses now return a validation error instead of reaching the mapper and producing a 500.
- The same-timestamp page-boundary regression pins canonical PostgreSQL UUID ordering. Empirical verification found `.NET Guid.CompareTo` agrees with that order; the mixed-endian default `Guid.ToByteArray()` is the actual ordering hazard.

## Summary by CodeRabbit

* **New Features**
  * Added an admin-only withdrawal CSV export endpoint with filtering by ID, user/completer, amount range, status, and date window.
  * CSV exports now include a new `receipt_key` column (alongside expiring receipt download links).
  * Large exports stream as RFC-compliant CSV using cursor-based paging.
* **Bug Fixes**
  * Improved export query validation with shared filter rules, stricter status checks, and corrected fee/net behavior for legacy “completed without payout” cases.
  * Export streaming is cancellation-safe and returns complete/consistent CSV results.
* **Chores**
  * Added a database index on `(CreatedAt, Id)` to optimize export pagination.

### CODE REVIEW

File: App/Modules/Withdrawals/Data/WithdrawalRepository.cs [hunk 1]
```diff
@@ -62,6 +38,83 @@ public async Task<Result<IEnumerable<WithdrawalPrincipal>>> Search(WithdrawalSea
...
+      var result = await query
+        .Include(x => x.Wallet)
+        .ThenInclude(x => x.User)
+        .Include(x => x.Completer)
+        .Include(x => x.Refunds)
+        .AsSplitQuery()
+        .OrderBy(x => x.CreatedAt)
+        .ThenBy(x => x.Id)
+        .Take(limit)
+        .ToArrayAsync(cancellationToken);
```
[the remark this answers is not in the document]
Comment (adelphi-liong) [reply App/Modules/Withdrawals/Data/WithdrawalRepository.cs:67]: Verified against the exact Npgsql.EntityFrameworkCore.PostgreSQL 8.0.2 provider used here with a standalone ToQueryString() probe. This expression translates to native SQL: WHERE "CreatedAt" > @p0 OR ("CreatedAt" = @p0 AND "Id" > @p1), followed by ORDER BY "CreatedAt", "Id". The predicate and ordering therefore both use PostgreSQL uuid ordering. The fake likewise uses .NET ordering for both operations, so it is internally consistent. The proposed x.Id > cursor.Id replacement is not valid C# because System.Guid does not define relational operators. Keeping the verified CompareTo expression.

By Codex 🤖

[the remark this answers is not in the document]
Comment (adelphi-liong) [reply App/Modules/Withdrawals/Data/WithdrawalRepository.cs:67]: Rechecked after the evidence reply and there is no counterexample. The exact provider translates this predicate to native PostgreSQL uuid comparison, while the suggested Guid relational operator is not valid C#. Closing this verified false positive.

By Codex 🤖

File: App/Modules/Withdrawals/Data/WithdrawalStorage.cs [hunk 2]
```diff
@@ -14,6 +14,11 @@ public Task<Result<string>> Save(Stream stream)
 
   public Task<Result<string>> Get(string key)
   {
-    return file.SignedLink(BlockStorages.Main, key, 60 * 60);
+    return this.Get(key, TimeSpan.FromHours(1));
+  }
+
+  public Task<Result<string>> Get(string key, TimeSpan expiry)
+  {
+    return file.SignedLink(BlockStorages.Main, key, (int)expiry.TotalSeconds);
```
[the remark this answers is not in the document]
Comment (adelphi-liong) [reply App/Modules/Withdrawals/Data/WithdrawalStorage.cs:36]: Fixed in 20ae6e2. The explicit-expiry overload now rejects values below one second or above seven days as a failed Result containing ArgumentOutOfRangeException before the MinIO client is called. Added direct tests for negative, zero, sub-second, over-limit, and both inclusive boundary values; the withdrawal suite is 206/206.

By Codex 🤖
